magento/pwa-studio
# feat(targets): Add Payment target

## Description
As developer, we want to build our own Payment methods as Extension for that reasons.
Payments should be extendable via Targetables


## Related Issue
Closes #2816

## Acceptance

### Verification Stakeholders

### Specification



### Verification Steps
1. Checkout should work as before 
2.  Scaffolding
```
DEBUG_PROJECT_CREATION=true ./pwa-studio/packages/pwa-buildpack/bin/buildpack create-project payment-test-1 --template "venia-concept" --name "payment-test-1" --author "larsroettig" --backend-url "https://master-7rqtwti-mfwmkrjfqvbjk.us-4.magentosite.cloud/" --braintree-token "sandbox_8yrzsvtm_s2bg8fs563crhqzk" --npm-client "yarn"
```
3. Payment addable via 
payment-test-1/local-intercept.js
```js 
function localIntercept(targets) {
    targets.of("@magento/venia-ui").payments.tap((payments) =>
        payments.add({
            paymentCode: "custome",
            importPath: "src/components/MyPayment/payment.js",
        })
    );
}
```

## Screenshots / Screen Captures (if appropriate)

![Screenshot at 11-50-24](https://user-images.githubusercontent.com/5289370/97078859-82c13800-15ef-11eb-9a0d-c772964276cf.png)


## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
